src/Draco.Fuzzing/Fuzzer.cs

```diff
@@ -68,17 +70,23 @@ private sealed class QueueEntry(TInput input, ExecutionResult? executionResult =
     /// </summary>
     public required ITracer<TInput> Tracer { get; init; }
 
-    private readonly Queue<QueueEntry> inputQueue = new();
-    private readonly HashSet<TCoverage> seenCoverages = [];
+    /// <summary>
+    /// True if the fuzzer should run in multithreaded mode. Only recommended for out-of-process execution.
+    /// </summary>
+    public bool Multithread { get; init; }
+
+    private readonly BlockingCollection<QueueEntry> inputQueue = [];
+    private readonly ConcurrentHashSet<TCoverage> seenCoverages = [];
+    private readonly object tracerSync = new();
 
     /// <summary>
     /// Enqueues the given input into the fuzzer.
     /// </summary>
     /// <param name="input">The input to enqueue.</param>
     public void Enqueue(TInput input)
     {
-        this.inputQueue.Enqueue(new QueueEntry(input));
-        this.Tracer.InputsEnqueued([input]);
+        this.inputQueue.Add(new QueueEntry(input));
+        lock (this.tracerSync) this.Tracer.InputsEnqueued([input]);
```

Comment:
Why not making the tracer thread safe ?

Reply:
Both a console and a TUI tracer would essentially need to synchronize with the main/rendering thread anyway

Reply:
The compiler will eventually parallelize compiling multiples files no ?

Reply:
Yes. But this has nothing to do with that, this is just the "observer" type for the fuzzer so the consumer can render it in a console or something

Reply:
Just to clarify, this tracer type is shipped by the fuzzer lib, not by the compiler, and is called by the fuzzer

Reply:
oh ok ok